torchcfm/conditional_flow_matching.py

```diff
@@ -54,7 +54,14 @@ def __init__(self, sigma: float = 0.0):
         ----------
         sigma : float
         """
-        self.sigma = sigma
+        self._sigma = sigma
+
+    @property
+    def sigma(self):
+        if isinstance(self._sigma, int):
+            return float(self._sigma)
+        else:
+            return self._sigma
```

Comment:
I don't think we should do this for reasons I've explained -- I think it is confusing. If we need sigma to always be a float we should typecast/typecheck in the init, or, we should write the logic such that for any input, the typing is conserved.

Reply:
I just changed it, thanks for pointing it out.